buddyboss/buddyboss-platform
# fix/3090 - Add/fix object caching for dozens of objects, reducing redundant queries significantly

Signed-off-by: nixsee <nick_chomey@hotmail.com>

### All Submissions:

* [x ] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [ x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x ] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?


### Changes proposed in this Pull Request:

Various functions try to cache data, but about 20 cache groups that they refer to were never created. It would be nearly impossible to estimate how many duplicate queries are performed as a result of these caching groups not existing.

Also added a cache group and caching mechanism for the groups_get_user_groups() and get_group_ids () functions, which are called 17 different times in 15 different files, including 28 times in each network search, resulting in an enormous amount of duplicate database queries.

Fixes #3090 

### How to test the changes in this Pull Request:

1. Do a global search for the cache groups that have been added to confirm that functions are trying to use them
2. Install Query Monitor and do a network search. Look at the duplicate queries to see 28 (14 of two types) duplicate queries related to getting a user's groups. 
3. Apply these code changes and see that the duplicate queries disappear. 

### Proof Screenshots or Video

![image](https://user-images.githubusercontent.com/88559987/138317010-e311977d-1b5f-4845-8f29-49c0d9dc941f.png)

![image](https://user-images.githubusercontent.com/88559987/138317020-515840d7-89f8-4e30-8ced-c57bf5007059.png)


### Other information:

* [ x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ x] Have you written new tests for your changes, as applicable?
* [x ] Have you successfully run tests with your changes locally?



### Changelog entry

Added/fixed object caching for many BB components, dramatically reducing redundant database queries.
